server/src/main/scala/de/frosner/broccoli/signal/SignalManager.scala

```diff
@@ -0,0 +1,24 @@
+package de.frosner.broccoli.signal
+
+import sun.misc.{Signal, SignalHandler}
+
+/**
+  * Provide a way to register and unregister OS signals
+  */
+trait SignalManager {
+
+  /**
+    * Registers signal handler for the given signal
```

Comment:
Minor nitpick: Please add a punctuation at the end, and a blank line below the first line.

In Scaladoc the first line serves as a summary that you see in the overview list of all methods. This only works if there's a blank line after it, and it reads better with proper punctuation.